ironfish/src/rpc/adapters/httpAdapter.ts

```diff
@@ -87,6 +88,11 @@ export class RpcHttpAdapter implements IRpcAdapter {
   }
 
   start(): Promise<void> {
+    if (this.started) {
+      return Promise.resolve()
+    }
+
+    this.started = true
```

Comment:
Should this line be at the bottom of the method? If it fails mid method the adapter will still look started.

Reply:
That's a good question. There is a trade off with both. Having it here solves re-entrancy issues with start(). Calling start() multiple times is safe. In the reverse, if an exception is thrown in start() then the adapter crashes then this instance is busted. I think that's a fine tradeoff because it will crash the program anyway and the adapter won't be usable.